bson/unmarshal.go

```diff
@@ -61,6 +61,21 @@ func UnmarshalWithContext(dc bsoncodec.DecodeContext, data []byte, val interface
 	return unmarshalFromReader(dc, vr, val)
 }
 
+// UnmarshalValue parses the BSON value of type t with bson.DefaultRegistry and
+// stores the result in the value pointed to by val. If val is nil or not a pointer,
+// UnmarshalValue returns an error.
+func UnmarshalValue(t bsontype.Type, data []byte, val interface{}) error {
+	return UnmarshalValueWithRegistry(DefaultRegistry, t, data, val)
+}
+
+// UnmarshalValueWithRegistry parses the BSON value of type t with registry r and
+// stores the result in the value pointed to by val. If val is nil or not a pointer,
+// UnmarshalValue returns an error.
+func UnmarshalValueWithRegistry(r *bsoncodec.Registry, t bsontype.Type, data []byte, val interface{}) error {
```

Comment:
See comment on UnmarshalValue.

Reply:
We should remove this API because we are deprecating all `*WithRegistry` marshaling functions with plans to remove them in Go Driver 2.0. Instead, we should provide `Encoder.MarshalValue` and `Decoder.UnmarshalValue` methods. We can add those in this PR or in a subsequent PR.

Reply:
What is the plan for the `*WithRegistry`? How do you think keeping the `UnmarshalValueWithRegistry` in this PR to keep the code consistent before a subsequent `*WithRegistry` PR?

Reply:
Our plan is to deprecate all `*WithRegistry` functions and recommend that users use a `bson.Encoder` or `bson.Decoder` instead if they need to use a custom `Registry` (see [GODRIVER-2711](https://jira.mongodb.org/browse/GODRIVER-2711)). However, for symmetry of the current API, we can keep this with the understanding that it will be deprecated soon after it's added.